src/mapeo-project.js

```diff
@@ -501,6 +504,104 @@ export class MapeoProject {
+    }
+    const zip = await yauzl.open(configPath)
+    // check for already present presets and delete them if exists
+    // TODO: do the same for fields and icons?
+    const ids = (await this.preset.getMany()).map((doc) => doc.versionId)
+    if (ids.length !== 0) {
+      console.log('already present', ids)
+      await this.preset.delete(ids)
+    }
+    /** @type {Object<string,{name:string,
+     * variants:Array<{
+     * size:string,
+     * mimeType: string,
+     * pixelDensity: number,
+     * blob: Buffer
+     * }>}>}
+     * */
+    const icons = {}
+    let presets = {}
+    /** @type {Object<String,String>} */
+    let fieldIds = {}
+    /** @type {Object<String,String>} */
+    let iconIds = {}
+    try {
+      const entries = await zip.readEntries()
+      for await (const entry of entries) {
+        // 1. for each icon, get the variant object and add it to icons
+        if (entry.filename.match(/icons\/[aA-zZ]/)) {
+          const icon = await parseIcon(
+            entry.filename.split('/')[1],
+            await zip.openReadStream(entry)
+          )
+          if (icon) {
+            if (!icons[icon.name]) {
+              icons[icon.name] = { name: icon.name, variants: [icon.variant] }
+            } else {
+              // icon already exists, so we push the variants
+              icons[icon.name].variants.push(icon.variant)
+            }
+          }
+        }
+        if (entry.filename === 'presets.json') {
+          // 2. read fields and presets from file
+          const presetsFile = await readPresets(await zip.openReadStream(entry))
+          presets = presetsFile.presets
+
+          // 3. For each field, add it to the fields db and get the id
+          for (let [fieldName, field] of Object.entries(presetsFile.fields)) {
+            const { docId, tagKey } = await addField({
+              fieldName,
+              field,
+              fieldDb: this.field,
+            })
+            fieldIds[tagKey] = docId
+          }
+        }
+      }
+    } catch (e) {
+      console.log('error parsing file on config zip', e)
+    } finally {
+      // 4. iterate over icons and add them to the db
+      /* eslint-disable no-unused-vars */
+      for (let [_, icon] of Object.entries(icons)) {
+        iconIds[icon.name] = await this.#iconApi.create(icon)
+      }
+      // 5. for each preset get the corresponding fieldId and iconId, add them to the db
+      for (let [presetName, preset] of Object.entries(presets)) {
+        preset.schemaName = 'preset'
+        preset.fieldIds = []
+        for (let field of preset.fields || []) {
+          preset.fieldIds.push(fieldIds[field])
+        }
+        delete preset.fields
+        preset.iconId = iconIds[preset.icon]
+        delete preset.icon
+        preset.name = presetName
+        // TODO: this is ugly
+        preset.addTags = {}
+        preset.removeTags = {}
+        if (!preset.terms) preset.terms = []
```

Comment:
sometimes `preset.terms` doesn't exist, hence the check. We could add it to the default config to every preset

Reply:
I think ok to add it.